fintech-ui/src/app/login/login.component.html

```diff
@@ -0,0 +1,45 @@
+<form (ngSubmit)="onSubmit()" [formGroup]="loginForm">
+  <div class="card">
+    <div class="card-body">
+      <h4 class="card-title">Login</h4>
+      <div class="form__margin">
+        <div class="form-group">
+          <input
+            [ngClass]="{ 'is-invalid': loginForm.get('username')?.errors && loginForm.get('username')?.dirty }"
```

Comment:
**Suggestion:**

- to minimise cumbersome code you may use form controls getters in your login.component.ts like 

> get username() { return this.loginForm.get('username'); }

- then you could reuse it in login.component.html 

> *ngIf="username?.errors?.required && username?.dirty"